File: app/src/main/java/co/tpcreative/supersafe/common/request/SyncItemsRequest.java
package co.tpcreative.supersafe.common.request;
import com.google.gson.Gson;

import java.io.Serializable;
import co.tpcreative.supersafe.R;
import co.tpcreative.supersafe.common.services.SuperSafeApplication;
import co.tpcreative.supersafe.model.DriveEvent;
import co.tpcreative.supersafe.model.Items;

public class SyncItemsRequest implements Serializable {

    public String user_id;
    public String device_id;
    public long size;
    public String cloud_id;
    public String kind;
    public String items_id;
    public String name;
    public String mimeType;
    public String global_original_id;
    public String global_thumbnail_id;
    public String categories_id;
    public boolean isSyncCloud;
    public boolean isSyncOwnServer;
    public boolean thumbnailSync;
    public boolean originalSync;
    public boolean isDeleteLocal;
    public boolean isDeleteGlobal;
    public String originalName;
    public String thumbnailName;
    public int formatType;
    public String thumbnailPath;
    public String originalPath;
    public String fileExtension;
    public int statusAction;
    public int degrees;
    public int fileType;
    public String title;
    public String nextPage;

    /*Sync items*/
    public SyncItemsRequest(String user_id,String cloud_id,String device_id,Items items){
        this.user_id = user_id;
        this.cloud_id = cloud_id;
        this.device_id = device_id;
        this.size = Long.parseLong(items.size);
        this.kind = SuperSafeApplication.getInstance().getString(R.string.key_drive_file);
        this.items_id = items.items_id;
        this.mimeType = items.mimeType;
        this.global_original_id = items.global_original_id;
        this.global_thumbnail_id = items.global_thumbnail_id;
        this.categories_id = items.categories_id;
        this.isSyncCloud = items.isSyncCloud;
        this.isSyncOwnServer = items.isSyncOwnServer;
        this.thumbnailSync = items.thumbnailSync;
        this.originalSync = items.originalSync;
        this.isDeleteLocal = items.isDeleteLocal;
        this.isDeleteGlobal = items.isDeleteGlobal;
        this.originalName = items.originalName;
        this.thumbnailName = items.thumbnailName;
        this.formatType = items.formatType;
        this.thumbnailPath = items.thumbnailPath;
        this.originalPath = items.originalPath;
        this.fileExtension = items.fileExtension;
        this.statusAction = items.statusAction;
        this.degrees = items.degrees;
        this.fileType = items.fileType;
        this.title = items.title;
        DriveEvent contentTitle = new DriveEvent();
        contentTitle.items_id = items.items_id;
        String hex = DriveEvent.getInstance().convertToHex(new Gson().toJson(contentTitle));
        this.name = hex;
    }

    /*Delete item*/
    public SyncItemsRequest(String user_id,String cloud_id,String items_id){
        this.user_id = user_id;
        this.cloud_id = cloud_id;
        this.items_id = items_id;
    }

    /*Get items list*/
    public SyncItemsRequest(String user_id,String cloud_id,String device_id,boolean isSyncCloud, String nextPage){
        this.user_id = user_id;
        this.cloud_id = cloud_id;
        this.isSyncCloud = isSyncCloud;
        this.nextPage = nextPage;
    }
}

File: app/src/main/java/co/tpcreative/supersafe/common/request/SyncItemsRequest.kt
package co.tpcreative.supersafe.common.request
import co.tpcreative.supersafe.R
import co.tpcreative.supersafe.common.services.SuperSafeApplication
import co.tpcreative.supersafe.common.util.Utils
import co.tpcreative.supersafe.model.DriveEvent
import co.tpcreative.supersafe.model.ItemModel
import co.tpcreative.supersafe.model.User
import com.google.gson.Gson
import java.io.Serializable

class SyncItemsRequest : Serializable {
    var user_id: String?
    var device_id: String?
    var size: Long = 0
    var cloud_id: String?
    var kind: String? = null
    var items_id: String? = null
    var name: String? = null
    var mimeType: String? = null
    var global_original_id: String? = null
    var global_thumbnail_id: String? = null
    var categories_id: String? = null
    var isSyncCloud = false
    var isSyncOwnServer = false
    var thumbnailSync = false
    var originalSync = false
    var isDeleteLocal = false
    var isDeleteGlobal = false
    var originalName: String? = null
    var thumbnailName: String? = null
    var formatType = 0
    var thumbnailPath: String? = null
    var originalPath: String? = null
    var fileExtension: String? = null
    var statusAction = 0
    var degrees = 0
    var fileType = 0
    var title: String? = null
    var nextPage: String? = null

    /*Sync items*/
    constructor(user_id: String?, cloud_id: String?, device_id: String?, items: ItemModel) {
        this.user_id = user_id
        this.cloud_id = cloud_id
        this.device_id = device_id
        size = items.size!!.toLong()
        kind = SuperSafeApplication.getInstance().getString(R.string.key_drive_file)
        items_id = items.items_id
        mimeType = items.mimeType
        global_original_id = items.global_original_id
        global_thumbnail_id = items.global_thumbnail_id
        categories_id = items.categories_id
        isSyncCloud = items.isSyncCloud
        isSyncOwnServer = items.isSyncOwnServer
        thumbnailSync = items.thumbnailSync
        originalSync = items.originalSync
        isDeleteLocal = items.isDeleteLocal
        isDeleteGlobal = items.isDeleteGlobal
        originalName = items.originalName
        thumbnailName = items.thumbnailName
        formatType = items.formatType
        thumbnailPath = items.getThumbnail()
        originalPath = items.getOriginal()
        fileExtension = items.fileExtension
        statusAction = items.statusAction
        degrees = items.degrees
        fileType = items.fileType
        title = items.title
        val contentTitle = DriveEvent()
        contentTitle.items_id = items.items_id
        val hex: String? = DriveEvent.getInstance()?.convertToHex(Gson().toJson(contentTitle))
        name = hex
    }

    /*Delete item*/
    constructor(user_id: String?, cloud_id: String?, items_id: String?) {
        this.user_id = user_id
        this.cloud_id = cloud_id
        this.items_id = items_id
        device_id = SuperSafeApplication.getInstance().getDeviceId()
    }

    /*Delete item from system*/
    constructor(item: ItemModel?) {
        this.user_id = Utils.getUserId()
        this.cloud_id = Utils.getUserCloudId()
        this.items_id = item?.items_id
        device_id = SuperSafeApplication.getInstance().getDeviceId()
    }

    /*Get items list*/
    constructor(user_id: String?, cloud_id: String?, device_id: String?, isSyncCloud: Boolean, nextPage: String?) {
        this.user_id = user_id
        this.cloud_id = cloud_id
        this.device_id = device_id
        this.isSyncCloud = isSyncCloud
        this.nextPage = nextPage
    }

    /*Get Item List*/
    constructor(nextPage: String? = "0"){
        val mUser = Utils.getUserInfo()
        this.user_id = mUser?.email
        this.cloud_id = mUser?.cloud_id
        this.device_id = SuperSafeApplication.getInstance().getDeviceId()
        this.isSyncCloud = true
        this.nextPage = nextPage
    }
}